components/portal/main/services.js

```diff
@@ -64,20 +64,78 @@ define(['angular'], function(angular) {
     };
 
       /**
-       * set the frame title using theme
+       * set the window title
+       *
+       * results in title
+       * page-title | app-title | portal-title (in an application), or
+       * page-title | portal-title (if the app has same name as the portal), or
+       * app-title | portal-title (if page name unknown or redundant)
+       * portal-title (if page name unknown or redundant and app name redundant)
+       *
+       * Examples:
+       * "Timesheets | STAR Time Entry | MyUW" ,
+       * for the Timesheets page in an app named "STAR Time Entry" in
+       * a portal named "MyUW", or
+       *
+       * "Notifications | MyUW",
+       * for the notifications page in an app named "MyUW" in
+       * a portal named "MyUW".
+       *
+       * "STAR Time Entry | MyUW"
+       * in an app named "STAR Time Entry" in a portal named "MyUW"
+       * when the page name is unspecified.
+       *
+       * "MyUW"
+       * in an app named "MyUW" in a portal named "MyUW"
+       * when the page name is unspecified.
        */
-      function setTitle() {
-        var frameTitle = '';
-        if ($rootScope.portal && $rootScope.portal.theme) {
-          frameTitle = $rootScope.portal.theme.title;
-          if (frameTitle !== NAMES.title && !APP_FLAGS.isWeb) {
-            frameTitle = ' | ' + frameTitle;
-          } else {
-            // since frame title equals the title in NAMES lets not duplicate it
-            frameTitle = '';
+      function setTitle(pageTitle) {
+        var windowTitle = ''; // we finally set the window title to this.
+
+        // first, gather pieces of the desired window title
+
+        var portalTitle = ''; // the name of the portal if we can determine it
+        var appTitle = ''; // the name of the app
+
+        if ($rootScope.portal && $rootScope.portal.theme &&
+              $rootScope.portal.theme.title) {
```

Comment:
if we want to do something with the theme title, we should pass it in via parameters. `$rootScope` or even `$scope` shouldn't bleed out to services